lib/api_repo.js

```diff
@@ -231,14 +231,24 @@ ApiRepo.prototype.buildPackages =
                 cleanName.replace(new RegExp('-', 'g'), '/');
             opts.packageInfo.api.name = that.pkgPrefix + cleanName;
             opts.packageInfo.api.version = version;
-            opts.packageInfo.api.buildCommonProtos =
-                that.opts.buildCommonProtos;
+            if (version) {
+              opts.packageInfo.api.fullName =
+                  opts.packageInfo.api.name + '-' + version;
+            } else {
+              opts.packageInfo.api.fullName = opts.packageInfo.api.name;
+            }
             var semver = opts.packageInfo.api.semver[l];
             if (semver) {
               /* eslint-disable camelcase */
               opts.packageInfo.api.semantic_version = semver;
               /* eslint-enable camelcase */
             }
+            if (that.opts.protoGenPackageDeps) {
+              opts.packageInfo.api.protoGenPackageDeps =
+                  that.opts.protoGenPackageDeps.map(function(dep) {
+                    return that.pkgPrefix + dep;
```

Comment:
This part is probably language-specific. For example, the common library for Python and Ruby is called as 'googleapis-common-protos' for some reasons (see https://pypi.python.org/pypi/googleapis-common-protos/, rubygems.org/gems/googleapis-common-protos). 

Considering that, can you create a method instead of an inline function? Like,
```javascript
that.opts.protoGenPackageDeps.map(that.getPackageDepName.bind(that, l, dep));
  :
  :
ApiRepo.prototype.getPackageDepName = function(language, dep) {
  // TODO: support other languages.
  return this.pkgPrefix + dep;
}
```